dynamiqs/sesolve/sesolve.py

```diff
@@ -88,9 +88,9 @@ def sesolve(
     """  # noqa: E501
     # === convert arguments
     H = _astimearray(H)
-    psi0 = jnp.asarray(psi0, dtype=cdtype())
+    psi0 = _asqarray(psi0)
```

Comment:
Does this not break diffrax?

Reply:
No it's still a `PyTree` so diffrax handles it well it seems. But regarding the static `dims` there could be some subtelty in this regard. I suspect diffrax is promoting all the PyTree leafs to the same dtype or something like that.